### PR TITLE
Genetics backend, ( mutations to come another PR )

### PR DESCRIPTION
**Pull request by Bod9001**
tthis is all the backend stuff plus UI
what needs to be implemented now is mutations and code review

### Changelog:
CL: [New] You can now modify People's customisations with genetics.

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+using System.Linq;
+using UI.Core.NetUI;
+using UI.Objects.Medical;
+using UnityEngine;
+
+public class SudokuPuzzleGame : MonoBehaviour
+{
+
+	//TODO At countdown timer for check validity
+	//TODO Add unlock, Set up with Species stuff
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs]: Still TODO?

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationUnlockMiniGame.cs [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+			if (OtherElement.SatisfiesTarget() == false)
+			{
+				Satisfies = false;
+			}
+
+		}
+
+		if (Satisfies)
+		{
+			Logger.LogError("AAAAAAAAAA TODO EGGGGG");
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationUnlockMiniGame.cs]: Log

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+
+
+	public Dictionary<MutationMiniGameElement, float> AccumulatedForces =
+		new Dictionary<MutationMiniGameElement, float>();
+
+	public void MainSliderChangeMaster(float Value )
+	{
+
+		Logger.LogWarning(" Target >  " +SliderParameters.TargetLever + "  Actual value >  " + Value.ToString());
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs]: Log

File: UnityProject/Assets/Scripts/UI/Objects/Medical/DNASpeciesElement.cs [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+
+	public Transform WindowPanel;
+
+
+	public void GenerateOption(string BodyPartName)
+	{
+		if (GUI_DNAConsole != null && GUI_DNAConsole.IsMasterTab)
+		{
+			//Look up
+			Logger.LogError(BodyPartName);
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/DNASpeciesElement.cs]: Log

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs [hunk 5]
```diff
@@ -340,20 +344,24 @@ public void BodyPartRemoveHealthMaster()
 		/// <summary>
 		/// Server only - Tries to remove a body part
 		/// </summary>
-		public void TryRemoveFromBody(bool beingGibbed = false)
+		public void TryRemoveFromBody(bool beingGibbed = false, bool CausesBleed = true, bool Destroy = false, bool PreventGibb_Death = false)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs:347]: This entire change seems like it needs to be logged or explained.

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+{
+
+	public override Mutation GetMutation(BodyPart BodyPart,MutationSO _RelatedMutationSO)
+	{
+		return new InBigStomachMutation(BodyPart,_RelatedMutationSO);
+	}
+
+	public class InBigStomachMutation : Mutation
+	{
+		public InBigStomachMutation(BodyPart BodyPart,MutationSO _RelatedMutationSO) : base(BodyPart,_RelatedMutationSO)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs:17]: wot? what does this or do what is its use?

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		{
+
+		}
+
+		public override void SetUp()
+		{
+			var Stomach = BodyPart.GetComponent<Stomach>();
+			Stomach.StomachContents.SetMaxCapacity(99); //idk Custom thing, if it's preset custom
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs]: magic number

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+		}
+
+		public override void Remove()
+		{
+			var Stomach = BodyPart.GetComponent<Stomach>();
+			Stomach.StomachContents.SetMaxCapacity(2);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs]: another magic number

File: UnityProject/Assets/Scripts/HealthV2/Living/MutationsSO.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
+using System.Collections;
+using System.Collections.Generic;
+using HealthV2;
+using UnityEngine;
+
+[CreateAssetMenu(fileName = "_DoNotUse", menuName = "ScriptableObjects/Mutations/_DoNotUse")]
+public class MutationSO : ScriptableObject
+{
+	[Range(0, 100)] public int ResearchDifficult;
+	public int Stability = 0;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/HealthV2/Living/MutationsSO.cs:14]: would be cool if we got tooltips explaining these

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+	public void Start()
+	{
+		Stacking = this.GetComponent<Stackable>();
+	}
+
+	public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
+	{
+		if (DefaultWillInteract.Default(interaction, side) == false) return false;
+		if (interaction.TargetObject == gameObject) return false;
+		if ( Validations.HasComponent<DNAConsole>(interaction.TargetObject) == false) return false;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs:19]: `return Validations.HasComponent<DNAConsole>(interaction.TargetObject)`

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+		return true;
+	}
+
+
+	public void ServerPerformInteraction(PositionalHandApply interaction)
+	{
+
+		var DNAConsole = interaction.TargetObject.GetComponent<DNAConsole>();
+		if (DNAConsole != null)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs:28]: avoid nesting

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs [hunk 12]
```diff
@@ -0,0 +1,108 @@
+using System.Collections;
+using System.Collections.Generic;
+using Systems.MobAIs;
+using UnityEngine;
+
+public class MutationGrabber : MonoBehaviour, IExaminable , ICheckedInteractable<PositionalHandApply>
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:6]: I just noticed, why are these things not namespaced?

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+{
+
+	public List<MutationSO> CarryingMutations;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs]: `public List<MutationSO> CarryingMutations = new List<MutationSO>();`

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+	}
+
+
+	public void ServerPerformInteraction(PositionalHandApply interaction)
+	{
+		var  DinosaurLivingMutationCarrier = interaction.TargetObject.GetComponent<DinosaurLivingMutationCarrier>();
+
+		if (DinosaurLivingMutationCarrier != null)
+		{
+			if (DinosaurLivingMutationCarrier.StageSynchronise ==
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs]: avoid nesting here as well

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+				DNAConsole.AddMutationOfficial(mutation);
+			}
+
+			Chat.AddExamineMsgFromServer(interaction.Performer, $" You plug-in the {this.gameObject.ExpensiveName()} Into the {interaction.TargetObject.ExpensiveName()} Transferring all the unknownMutations, And clearing the buffer ");
+			CarryingMutations.Clear();
+		}
+
+	}
+
+	string IExaminable.Examine(Vector3 worldPos)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:89]: private string

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationInjector.cs [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+	public List<DNAMutationData> DNAPayload = new List<DNAMutationData>();
+
+
+	public SpriteHandler SpriteHandler;
+
+	public bool WillInteract(PositionalHandApply interaction, NetworkSide side)
+	{
+		if (DefaultWillInteract.Default(interaction, side) == false) return false;
+		if (interaction.TargetObject == gameObject) return false;
+		if (Validations.HasComponent<LivingHealthMasterBase>(interaction.TargetObject) == false) return false;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationInjector.cs:17]: `return Validations.HasComponent<LivingHealthMasterBase>(interaction.TargetObject)`

File: UnityProject/Assets/Scripts/Managers/GameManager.cs [hunk 17]
```diff
@@ -145,7 +148,7 @@ public partial class GameManager : MonoBehaviour, IInitialise
 	public bool LoadedDirectlyToStation => loadedDirectlyToStation;
 
 	public bool QuickLoad = false;
-
+	public bool QuickJoinLoad = false;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Managers/GameManager.cs:151]: why was this renamed?

File: UnityProject/Assets/Scripts/Objects/Medical/Genetics/DinosaurLivingMutationCarrier.cs [hunk 18]
```diff
@@ -0,0 +1,92 @@
...
+	//bunch of sprites for different dinosaurs
+
+	//dinosaur gets spawned depending on difficulty of each mutation (E.G T Rex if it's got loads of epic mutations) only correlates 75% of time, just a troll
+	//Feeding mechanics
+	//growth stages
+
+
+	//that's what needs to be done
+
+	//Chasmosaurus?
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Medical/Genetics/DinosaurLivingMutationCarrier.cs]: what does this comment mean? this seems like its going to confuse people.

File: UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs [hunk 19]
```diff
@@ -0,0 +1,89 @@
...
+
+    //Chasmosaurus?
+
+
+
+
+	//Triceratops = Friendly
+	//Dimetrodon angelensis = Friendly
+	//troodon  = Friendly
+
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs:26]: spaacccinnnggg

File: UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+	//Stegosaurus = Neutral
+	//Brtachiosaurus = Neutral
+
+    // 300
+    // 200 = Tarbosaurus, Brtachiosaurus, SPINOSAURUS >
+    // 100 = Stegosaurus,Triceratops, saichania >
+    // 0 = raptor, Dimetrodon angelensis, troodon >
+
+
+    private IEnumerator countdown()
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs]: `Countdown()`

File: UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs [hunk 21]
```diff
@@ -0,0 +1,89 @@
...
+		    var  Settings =  BodyPartMutations.GetMutationRoundData(mutationSo);
+		    difficulty += Settings.ResearchDifficult;
+
+	    }
+
+	    //TODO Separate prefabs for hostile and non-hostile dinosaurs
+	    GameObject Dinosaur = null;
+
+
+	    if (difficulty > 200)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs:57]: why not use a switch statement?

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/Spawned/DNAStrandList.cs [hunk 22]
```diff
@@ -0,0 +1,40 @@
...
+
+
+		public DNAStrandElement AddElement(DNAMutationData.DNAPayload Payload, string target, DNAStrandElement.Location SetLocation)
+		{
+			var NewElement  = AddItem() as DNAStrandElement;
+			NewElement.SetValues(Payload,target, SetLocation );
+			return NewElement;
+		}
+
+		//public void RemoveItem(DNAStrandElement DNAStrandElement)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/Spawned/DNAStrandList.cs]: don't comment out code like this or codacy will scream at us

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/Spawned/DNAStrandList.cs [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+
+namespace UI.Core.Net.Elements.Dynamic.Spawned
+{
+	public class DNAStrandList : EmptyItemList
+	{
+
+
+		public DNAStrandElement AddElement(DNAMutationData.DNAPayload Payload, string target, DNAStrandElement.Location SetLocation)
+		{
+			var NewElement  = AddItem() as DNAStrandElement;
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/Spawned/DNAStrandList.cs:18]: I don't get this code, what does it do?

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs [hunk 24]
```diff
@@ -10,11 +10,20 @@ namespace UI.Core.NetUI
...
 		public UnityEvent ServerMethod;
 
 		public override void ExecuteServer(PlayerInfo subject)
 		{
 			ServerMethod.Invoke();
 		}
+
+		public void Awake()
+		{
+			Button = this.GetComponent<Button>();
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs:25]: would be cool if you add a null check to this just in case

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs:25]: or make it so that this component automatically adds the button component if its added in the inspector 
```cs
[RequireComponent(typeof (Button))]
```

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetClientSyncString.cs [hunk 25]
```diff
@@ -0,0 +1,9 @@
+using System.Collections;
+using System.Collections.Generic;
+using UI.Core.NetUI;
+using UnityEngine;
+
+public class NetClientSyncString : NetServerSyncString
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetClientSyncString.cs:6]: why is this a whole class in an entire script?

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetColorChanger.cs [hunk 26]
```diff
@@ -45,13 +45,13 @@ public class NetColorChanger : NetUIElement<Color>
...
 		public Graphic Element => element ??= GetComponent<Graphic>();
 		private Graphic element;
-		
+
 		public override void ExecuteServer(PlayerInfo subject) { }
 
 		/// <summary>
 		/// Server-only method for updating element (i.e. changing label text) from server GUI code
 		/// </summary>
-		public override void SetValueServer(Color value)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetColorChanger.cs:54]: "set value server" makes much more sense than MasterSetValue, why the rename?

File: UnityProject/Assets/Scripts/UI/Core/Net/Page/NetPageSwitcher.cs [hunk 27]
```diff
@@ -66,7 +66,7 @@ public override void Init()
 				}
 			}
 
-			if (MasterTab.IsServer && !StartInitialized)
+			if (containedInTab.IsMasterTab && !StartInitialized)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Core/Net/Page/NetPageSwitcher.cs]: don't use ! unless you're inversing something
```cs
StartInitialized == false
```

File: UnityProject/Assets/Scripts/UI/Objects/Medical/GUI_DNAConsole.cs [hunk 28]
```diff
@@ -0,0 +1,212 @@
...
+		}
+
+		public void GenerateDNA()
+		{
+			var DNAMutationData = new DNAMutationData();
+
+			DNAMutationData.BodyPartSearchString = target.Value;
+
+
+			for (int i = 0; i < Viewport.childCount; i++)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/GUI_DNAConsole.cs:63]: wouldn't it be better to do this in reverse just in-case any of it's elements changes while this is running?

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationChoiceElement.cs [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+		MutationSO = InMutationSO;
+		MutationUnlockMiniGame = InMutationUnlockMiniGame;
+		NetText_label.SetValue($"Difficulty 100/{BodyPartMutations.GetMutationRoundData(InMutationSO).ResearchDifficult}");
+	}
+
+	public void OnSelect()
+	{
+		MutationUnlockMiniGame.GenerateForMutation(MutationSO);
+
+
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationChoiceElement.cs]: why the empty space

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs [hunk 30]
```diff
@@ -0,0 +1,138 @@
...
+		AccumulatedForces.Clear();
+		locked = false;
+	}
+
+	public void Awake()
+	{
+		netServerSyncString.OnChange.AddListener(Setup);
+		Indicator.GetComponent<Slider>().onValueChanged.AddListener(ValueChangeSlider);
+
+		//PreviouslyValueLever = Indicator
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs]: here as well, don't comment out code unless you write an explanation as to why its there and is at the end of the comment so you dont trigger codacy

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs [hunk 31]
```diff
@@ -0,0 +1,138 @@
...
+	}
+
+	public bool SatisfiesTarget()
+	{
+		var TargetFloat = SliderParameters.TargetPosition / 100f;
+
+
+		var NearValue =Mathf.Abs( Mathf.Abs(Indicator.Element.value - TargetFloat ) - 1f);
+
+		if (0.05f > Mathf.Abs((NearValue - 1f)))
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs:79]: magic number

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs [hunk 32]
```diff
@@ -0,0 +1,138 @@
...
+		AccumulatedForces.Clear();
+		locked = false;
+	}
+
+	public bool SatisfiesTarget()
+	{
+		var TargetFloat = SliderParameters.TargetPosition / 100f;
+
+
+		var NearValue =Mathf.Abs( Mathf.Abs(Indicator.Element.value - TargetFloat ) - 1f);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs:77]: magic number

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs [hunk 33]
```diff
@@ -0,0 +1,138 @@
...
+		{
+			return true;
+		}
+
+		return false;
+	}
+
+
+
+	public Dictionary<MutationMiniGameElement, float> AccumulatedForces =
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs]: don't put variables in the middle of functions

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs [hunk 34]
```diff
@@ -0,0 +1,138 @@
...
+		new Dictionary<MutationMiniGameElement, float>();
+
+	public void MainSliderChangeMaster(float Value )
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs]: `float value`

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs [hunk 35]
```diff
@@ -0,0 +1,189 @@
...
+	public EmptyItemList SpeciesSelectionList;
+
+	public PlayerHealthData currentlySelected;
+	public BodyPartMutations.MutationRoundData currentlySelectedData;
+
+	public Dictionary<PlayerHealthData, SpeciesChoiceElement> MutationToElement =
+		new Dictionary<PlayerHealthData, SpeciesChoiceElement>();
+	public void Start()
+	{
+		if (GUI_DNAConsole.IsMasterTab)
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs]: can be simplifed
```cs
if (GUI_DNAConsole.IsMasterTab == false) return;
```

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs [hunk 36]
```diff
@@ -0,0 +1,189 @@
...
+			if (Show)
+			{
+				Status.SetValue("Invalid board incorrect state");
+			}
+
+			return false;
+		}
+		else
+		{
+			if (board.Contains("."))
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs:142]: same here, try not to nest stuff

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/sudoku generator/SudokuGenerator.cs [hunk 37]
```diff
@@ -0,0 +1,600 @@
...
+
+
+	public Dictionary<string, string> _get_candidates_map(List<char> board){
+		/* Get all possible candidates for each square as a map in the form
+		{square: sudoku.DIGITS} using recursive constraint propagation. Return `false`
+		if a contradiction is encountered
+		*/
+
+		// Assure a valid board
+		var report = validate_board(board);
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/sudoku generator/SudokuGenerator.cs:125]: spacing is messed up here

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/sudoku generator/SudokuGenerator.cs [hunk 38]
```diff
@@ -0,0 +1,600 @@
...
+
+        // Give up and try a new puzzle
+        return generate(InDIFFICULTY);
+    }
+
+
+	public Dictionary<string, string> _get_candidates_map(List<char> board){
+		/* Get all possible candidates for each square as a map in the form
+		{square: sudoku.DIGITS} using recursive constraint propagation. Return `false`
+		if a contradiction is encountered
```
Comment (MaxIsJoe) [inline UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/sudoku generator/SudokuGenerator.cs:121]: would be nice if we got deleted explanation like this everywhere :)
even better, some of these are best written in xml format above functions so we can read them in IDEs when we hover over them

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs:347]: What would you add?

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs (hunk 6 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs:17]: basically so forces you to use the constructor and pass in the variables

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs (hunk 7 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs]: was debug mutation

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs (hunk 8 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs]: going to think of a better system that resets it to the default stomach size

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs (hunk 10 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs:19]: nah,I feel like that's unneeded makes it less clear

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs (hunk 11 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/Amber.cs:28]: I feel like it's fine in this scenario

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs (hunk 12 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:6]: Just never did it

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs (hunk 5 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs:347]: This change can disable bleeding on some parts, destory them and can prevent death on gib; correct?

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs (hunk 6 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs:17]: ah, good then.

File: UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs (hunk 7 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/HealthV2/Living/Mutations/BigStomachMutation.cs]: this can still trigger codacy

File: UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs (hunk 5 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/HealthV2/Living/BodyParts/BodyPart.cs:347]: yeah?

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs (hunk 12 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:6]: namespace things, it will make whoever decides to try and setup assembly definitions for us's life much easier.

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs (hunk 1 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/SudokuPuzzleGame.cs]: yeah they Done

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs (hunk 13 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs]: technically unity does a automatic   new List<MutationSO>(); . But it's good to add

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs (hunk 15 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:89]: image

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationInjector.cs (hunk 16 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationInjector.cs:17]: same as above

File: UnityProject/Assets/Scripts/Managers/GameManager.cs (hunk 17 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Managers/GameManager.cs:151]: it's a new field doesn't replace the old one

File: UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs (hunk 21 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Objects/Medical/Genetics/GeneticEggLogic.cs:57]: I'm not sure how to do a > In the switch statement

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/Spawned/DNAStrandList.cs (hunk 23 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/Dynamic/Spawned/DNAStrandList.cs:18]: me trying to work out how Netlists work, turns out I don't need to have it, but already made it so Can't be bothered to move it,you think it's worth saying adding a comment that an empty list would just work is fine?

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs (hunk 24 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetButton.cs:25]: yeha it has it
namespace UI.Core.NetUI
{
	/// Simple button, has no special value
[RequireComponent(typeof(Button))]

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetColorChanger.cs (hunk 26 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetColorChanger.cs:54]: because when you host the game yourself there is technically two UIs, one is the one that is considered server/master
and then there's a client Emulated version one as well, so you can be on the server while still being on the client one

File: UnityProject/Assets/Scripts/UI/Core/Net/Page/NetPageSwitcher.cs (hunk 27 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Page/NetPageSwitcher.cs]: to be fair is old code

File: UnityProject/Assets/Scripts/UI/Objects/Medical/GUI_DNAConsole.cs (hunk 28 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Objects/Medical/GUI_DNAConsole.cs:63]: it's only generating a DNA payload it shouldn't change over the course, it's all it's doing is grabbing the data, and it's not doing any Await things

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs (hunk 31 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs:79]: You think it's worth pulling out the margin of error value?

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs (hunk 32 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/MutationMiniGameElement.cs:77]: the -1 is inverting the value so technically not magic

File: UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/sudoku generator/SudokuGenerator.cs (hunk 37 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Objects/Medical/genetics/SudokuPuzzle/sudoku generator/SudokuGenerator.cs:125]: it is because JS port

File: UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetClientSyncString.cs (hunk 25 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/UI/Core/Net/Elements/NetClientSyncString.cs:6]: makes it a bit more clear?

File: UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs (hunk 12 above)
Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:6]: ok cool

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Items/Medical/Genetics/MutationGrabber.cs:6]: humm, since I'm just auto generating the namespace for like they could do the same thing so I'm going to leave it for now
